tests/integration/security/sds_tls_origination/util/test_certs.go

```diff
@@ -84,3 +99,119 @@ AcCmGiv5+fUBaa/AgMDBD3YKxFRkEEx8ns2C/R2EGtSoNbLmoVxPqsc13/rVcR2b
 pNCxV9Zp0EA6qmvtb5A+tBo4BTRIZbAZ1jOZcjP4lOl+cA7FoNk6f9OE
 -----END RSA PRIVATE KEY-----`
 )
+
+// CreateRoleForSecretAccessInNamespace creates Roles in given namespace for listing secrets
+func CreateRoleForSecretAccessInNamespace(t framework.TestContext, roleName string, namespace string, clusters ...cluster.Cluster,
+) {
+	t.Helper()
+
+	// Create Kubernetes role for secret access
+	wg := multierror.Group{}
+	if len(clusters) == 0 {
+		clusters = t.Clusters()
+	}
+	for _, c := range clusters {
```

Comment:
this can be simplified a lot by using ApplyYaml. It handles all the cluster stuff, error checking, etc

Reply:
Also cleanup

Reply:
Done